### ISSUE TITLE
Can BlockOptimizer switch learning rates?

### ISSUE THREAD

**Issue by a-F1** [Ledzy/BAdam#11]
Thank you for your excellent work!

I have a quick question: When using the BlockOptimizer to switch blocks, is it possible to simultaneously switch other optimizer parameters, such as the learning rate and weight decay rate?

**Comment by Ledzy** [comment]
Hi, you can directly ajust these parameters after each block switching:

```python
# update weight_decay
optimizer.base_optimizer.param_groups[0]["weight_decay"] = new_weight_decay

# update lr
for param_group in optimizer.param_groups:
    param_group["lr"] = new_lr
```

**Comment by a-F1** [comment]
Thank you for your thorough and detailed response! I also appreciate your excellent work!
